### PR TITLE
feat: move closing chunks as soon as possible

### PR DESCRIPTION
**Pull request by tustvold**
Having chunks linger in a closing state seems to be a repeated source of confusion, and currently serves no purpose as the requisite logic is not hooked up in the write path anyway.

This PR changes the mutable_linger_seconds and mutable_minimum_age_seconds to only impact whether a chunk in the open state is considered a move candidate, any closing chunks will immediately be considered candidates, with the order determined by the sort order.

This effectively redefines the closing state to be the window of time between when a chunk exceeds the mutable size threshold and when it starts to be moved to the read buffer based on available resource (currently the heuristic for this is very basic, it just moves them one at a time)

### CODE REVIEW

**Review by mkmik — APPROVED** [review]

File: server/src/db/lifecycle.rs [hunk 1]
```diff
@@ -119,7 +119,7 @@ trait ChunkMover {
                     move_active = true;
                     self.move_to_read_buffer(partition_key, table_name, chunk_id);
                 }
-                ChunkState::Closing(_) if would_move => {
+                ChunkState::Closing(_) if !move_active => {
```
Comment (crepererum) [inline server/src/db/lifecycle.rs:122]: I'm wondering how we can arrive at this state? External API calls? The case above does open=>closing=>moving in one go, so the intermediate state shouldn't be visible.

Comment (tustvold) [reply server/src/db/lifecycle.rs:122]: https://github.com/influxdata/influxdb_iox/blob/main/server/src/db.rs#L944

Comment (crepererum) [reply server/src/db/lifecycle.rs:122]: Ohhhhhh, now it makes sooo much more sense. :exploding_head:

**Review by crepererum — APPROVED** [review]

File: server/src/db/lifecycle.rs (hunk 1 above)
Comment (crepererum) [reply server/src/db/lifecycle.rs:122]: We should probably move the entire life-cycle more into a single place (doesn't need to be a single loop, 1 module would already work).

**Review by alamb — APPROVED** [review]
I think this change makes sense from a end user point of view too

The main usecase of the linger time, as I understood it, was to catch stragglers across *partition* boundaries (so when new data was written into a new partition) and avoid creating tiny new chunks with a small number of rows  in a previously closed partition. 

When a previous chunk has rolled over and new writes are going into a new chunk in the same partition there is no reason for that old chunk to linger open

File: server/src/db/lifecycle.rs (hunk 1 above)
Comment (tustvold) [reply server/src/db/lifecycle.rs:122]: @crepererum Fortunately someone already split out the closing logic into a free function :grinning:

**Pull request merged**